### PR TITLE
 #167735044 Build Return Trip Endpoint

### PR DESCRIPTION
**Pull request by hameedhub**
**What does this PR do?**

- Have user request return trip

  **Description of Task to be completed?**

  Have user return trip endpoint working
- `POST /api/v1/request/return`

  **How should this be manually tested?**

 Clone the repo cd into it and RUN _npm run server_ 
- Using postman test the above endpoint with 
_key_ : Content-Type: _value_ : application/json
- To get authentication, Login or Signup and get TOKEN
_key_: Authorization _value_: TOKEN

  **Any background context you want to provide?**

  - None

  **What are the relevant stories?**

  https://www.pivotaltracker.com/story/show/167735044

  **Screenshots (if appropriate)**

 
Screen Shot 2019-08-30 at 3 03 44 AM

  **Questions:**

- None

### CODE REVIEW

File: src/validation/schemas.js [hunk 1]
```diff
@@ -22,3 +22,34 @@ export const signUpSchema = Joi.object().keys({
...
+    })),
+  destination: Joi.string().trim().required()
+    .error(() => ({
+      message: 'Destination is required'
+    })),
+  travelDate: Joi.string().regex(/^(?:(?:31(\/|-|\.)(?:0?[13578]|1[02]))\1|(?:(?:29|30)(\/|-|\.)(?:0?[13-9]|1[0-2])\2))(?:(?:1[6-9]|[2-9]\d)?\d{2})$|^(?:29(\/|-|\.)0?2\3(?:(?:(?:1[6-9]|[2-9]\d)?(?:0[48]|[2468][048]|[13579][26])|(?:(?:16|[2468][048]|[3579][26])00))))$|^(?:0?[1-9]|1\d|2[0-8])(\/|-|\.)(?:(?:0?[1-9])|(?:1[0-2]))\4(?:(?:1[6-9]|[2-9]\d)?\d{2})$/).required()
+    .error(() => ({
+      message: 'Invalid travel date format (e.g day/month/year)',
+    })),
+  returnDate: Joi.string().regex(/^(?:(?:31(\/|-|\.)(?:0?[13578]|1[02]))\1|(?:(?:29|30)(\/|-|\.)(?:0?[13-9]|1[0-2])\2))(?:(?:1[6-9]|[2-9]\d)?\d{2})$|^(?:29(\/|-|\.)0?2\3(?:(?:(?:1[6-9]|[2-9]\d)?(?:0[48]|[2468][048]|[13579][26])|(?:(?:16|[2468][048]|[3579][26])00))))$|^(?:0?[1-9]|1\d|2[0-8])(\/|-|\.)(?:(?:0?[1-9])|(?:1[0-2]))\4(?:(?:1[6-9]|[2-9]\d)?\d{2})$/).required()
```
Comment (elikeyz) [inline src/validation/schemas.js]: Use `Joi.date()` instead

**Review by prieste — COMMENTED** [review]
I notice you have traveldate and returnDate on your request body do we need both fields for a return trip

File: src/controllers/requestController.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+   *
+   * @param {object} req - request object
+   * @param {object} res - reponse object
+   * @returns {object} data - trip details
+   */
+  static async returnTrip({ body, userData }, res) {
+    try {
+      const data = await tripService.returnTrip({ ...body, ...userData });
+
+      res.status(201).json({ status: 'success', data });
```
Comment (prieste) [inline src/controllers/requestController.js]: there is a helper function for responses in the utils folder please utilise it

**Review by prieste — COMMENTED** [review]
I notice you have traveldate and returnDate on your request body do we need both fields for a return trip

File: src/routes/index.js [hunk 3]
```diff
@@ -1,9 +1,11 @@
 import { Router } from 'express';
 import authRoute from './auth';
+import tripRoute from './request';
 
 const router = Router();
 
 
 router.use('/auth', authRoute);
+router.use('/request', tripRoute);
```
Comment (prieste) [inline src/routes/index.js]: use /requests

File: src/controllers/requestController.js (hunk 2 above)
Comment (hameedhub) [reply src/controllers/requestController.js]: please consider refactoring the response class
Screen Shot 2019-09-01 at 2 31 44 PM

**Pull request merged**